### PR TITLE
feat: extended BaseCV class to support learning rate schedulers that need a metric to monitor

### PR DESCRIPTION
**Pull request by PabloNA97**
These changes are made to extend the available LR Schedulers to LR that monitor a metric like: ReduceLROnPlateau or OneCycleLR. 

I think that in these cases one needs to change the instance of the LR Scheduler class by a dictionary containing that instance and also some configurations. Including the metric to 'monitor', see the [Lightning manual](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.core.LightningModule.html#lightning.pytorch.core.LightningModule.configure_optimizers)

Hope it's useful :)

### CODE REVIEW

**Comment by luigibonati** [comment]
PS: In #227 I've added an example to the tutorial notebooks of the options to pass for such case and also logged the LR in the metrics in case a scheduler is used.

**Comment by PabloNA97** [comment]
> thanks for the contribution! and sorry for the delay in merging it :(

Please don't worry! Thanks to you for the mlcolvar library

I hope I can afford to make more contributions in the future :)
